qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1090,4 +1100,52 @@ private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 		}
 		return resolvedJavaType;
 	}
-}
+
+	private void validateSwitchWhenPart(JavaMethodInfo method, JavaMemberInfo member, Section ownerSection,
+			String projectUri, ResolvingJavaTypeContext resolvingJavaTypeContext, PropertyPart propertyPart,
+			MethodPart methodPart, List<Diagnostic> diagnostics) {
+		String javaTypeParent = "";
+		String partName = "";
+		if (method != null) {
+			javaTypeParent = method.getJavaElementType();
+			partName = methodPart.toString();
+		} else if (member != null) {
+			javaTypeParent = member.getJavaElementType();
+			partName = propertyPart.toString();
+		}
+		List<Node> children = ownerSection.getChildren();
+		// Iterate through children and ensure the type of the #case or #is parts are
+		// the same as the parent section
+		for (Node node : children) {
+			if (node.getKind() == NodeKind.Section) {
+				List<Parameter> value = new ArrayList<Parameter>();
+				if (ownerSection.getSectionKind() == SectionKind.SWITCH) {
+					value = ((CaseSection) node).getParameters();
+				} else if (ownerSection.getSectionKind() == SectionKind.WHEN) {
+					value = ((IsSection) node).getParameters();
+				}
+				for (Parameter child : value) {
+					String javaTypeChild = LiteralSupport.getLiteralJavaType(child.getValue());
+					// Case of Enum, check the Enum value is valid
+					if (javaTypeChild == null) {
+						String childName = child.getName();
+						ResolvedJavaTypeInfo javaTypeChildInfo = resolveJavaType(javaTypeParent, projectUri,
+								resolvingJavaTypeContext);
+						if (!"in".equals(childName)

```

Comment:
It seems that you don't take care of this usecase:

 * `{#is in}` : here `in` is the Java Type (not the in keyword).
`{#is in in on}`  : here the first `in` is the keyword and teh second  is the Java Type (not the in keyword).

Reply:
Could you add a sample in comment.